### DISCUSSION TITLE
How this all work?

### DISCUSSION THREAD

**Discussion by FernyHz** [LeleDerGrasshalmi/FortniteEndpointsDocumentation#145] [Q&A] [answered]
Hello. I'm new with all the endpoints. I just wanted to know how can I see the info of a user in stw using the Api (for example if he needs to do storm king) or if there is vbucks missions in fortnite stw. I was reading but I'm super novice with this. Is it possible to provide an example of what can I do to consume stw Api? Thanks in advance!!

**Comment by LeleDerGrasshalmi** [comment]
Hi,

StW Missions are in the [StW World API](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/d88f63a834aef8141bea7c4773a1ef053e3361bd/Fortnite/SaveTheWorld/Missions.md)

To query a Users StW Profile (campaign or common_public) you need to use the [`QueryPublicProfile` Profile MCP Operation](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/2cc783ec852eb65491ef55469a63c95be5a7ad50/Fortnite/MCP/operations/QueryPublicProfile.md), but first you should to read the [general MCP Profile API Infos](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/1ea5bb2b04f66a993a2541a9f39d189bfb8cf397/Fortnite/MCP/README.md)

**Comment by FernyHz** [comment]
Hey, 

Thanks for that info. I see this part "Auth Required: Yes" in mostly every endpoint. Can you tell me how can i get credentials for it or what do i have to do? Again thanks in advance!!

**Comment by entraptaa** [comment, marked as answer]
> Can you tell me how can i get credentials for it or what do I have to do? Again thanks in advance!!

Refer to the [Authentication Documentation](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/tree/main/EpicGames/AccountService/Authentication)

**Reply by FernyHz** [reply]
Hey, thanks for answering. I was reading it and I saw Auth Clients so im trying to get data from this endpoint [https://fngw-mcp-gc-livefn.ol.epicgames.com/fortnite/api/game/v2/world/info](url) and Im using as headers "client_id" and "client_secret" from the credentials table from EpicGames/AccountService/Authentication/Clients.md. but i'm getting Status 400 Bad request. So can you tell me If I'm using the headers in the right way or do i need to use basic auth. I'm sorry for asking a lot.

**Reply by LeleDerGrasshalmi** [reply]
Generate a Access Token:
https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/main/EpicGames/AccountService/Authentication/Token.md

**Reply by FernyHz** [reply]
Hey, thanks it worked. Now I was trying to see someone else profile in fortnite stw. and I was reading this Fortnite/MCP/README.md but I'm lost. Is it posible to have an example of http request of how to do that. I know i have to use campaign for stw data but how? Also where do i put the epic account name . I'm sorry for asking I'm just learning and I had to figure it out many things by myself. Thanks in advance again.

**Reply by LeleDerGrasshalmi** [reply]
Hey, sure heres an Example HTTP Message to query your campaign (main stw) profile

```http
POST /fortnite/api/game/v2/profile/ACCOUNT_ID/client/QueryProfile?profileId=campaign&rvn=-1 HTTP/1.1
Host: fngw-mcp-gc-livefn.ol.epicgames.com
Content-Type: application/json
Authorization: Bearer xxxxxxxxxxxxxx
Content-Length: 2

{}
```

This only works for your own Account, but you can use `QueryPublicProfile` to fetch anyones `campaign` and `common_public` profile.
To get the AcountId from the DisplayName checkout [the Displayname Lookup API](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/main/EpicGames/AccountService/Account/Lookup/Displayname.md)

**Reply by FernyHz** [reply]
Hey, yeah that worked with my account as you said. this is how I'm using it

> POST /fortnite/api/game/v2/profile/ACCOUNT_ID/client/QueryPublicProfile?profileId=common_public&rvn=-1
> Host: fngw-mcp-gc-livefn.ol.epicgames.com
> Authorization: Bearer XXXXX
> Content-Length: 2
> Content-Type: application/json
> {}

So I tried with different account_id but im getting this:
`
{ "errorCode": "errors.com.epicgames.common.missing_permission", "errorMessage": "Sorry your login does not posses the permissions 'fortnite:account_id:commands ALL' needed to perform the requested operation", "messageVars": [ "fortnite:profile:account_id:commands", "ALL" ], "numericErrorCode": 1023, "originatingService": "fortnite", "intent": "prod-live" }
`
Is it because of client credentials I'm using or Am I missing something? Thanks!!

**Reply by LeleDerGrasshalmi** [reply]
Its because QueryProfile csn ONLy be used for you OWN Profile (Account Authentication)

**Reply by FernyHz** [reply]
Oh okay. Also is there a way to get account id based on playstation display_name?

**Reply by LeleDerGrasshalmi** [reply]
Yes, using https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/main/EpicGames/AccountService/Account/Lookup/External/Displayname.md

**Reply by FernyHz** [reply]
Okay thanks. Also is it possible if you tell me what is this for [https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/d88f63a834aef8141bea7c4773a1ef053e3361bd/Fortnite/MCP/operations/UpdateQuests.md](url). Just in case does it have to do with like using it so you do your a random quest in a mission and then leave so it will count even if you didnt complete a mission in stw?

**Reply by LeleDerGrasshalmi** [reply]
Thats how the Server Updates your quest progress.

**Reply by FernyHz** [reply]
Thanks. okay so i was checking the missions [https://fngw-mcp-gc-livefn.ol.epicgames.com/fortnite/api/game/v2/world/info](url) for stw missions. Do you know if there is a pattern or what is the name of a vbuck mission?

**Comment by LeleDerGrasshalmi** [comment]
Gamefiles from the Mission generator

**Reply by FernyHz** [reply]
oh and do you know how can i access to it or if there is something i need to read for it?

**Reply by taradinoc** [reply]
https://github.com/BanjoByTheBay/BanjoBotAssets is useful for exporting the data you need to identify mission rewards.

**Comment by FernyHz** [comment]
Hey. I saw this [EpicGames/AccountService/Authentication/GrantTypes/refresh_token.md](url) and I would like to know if that refreshes the refresh_token so i can keep using the access_token. And will the refresh_token still get refreshed when using that endpoint after it expired(after 28800 seconds)?

**Reply by LeleDerGrasshalmi** [reply]
No, you get new tokens

[a comment by an automated account is not shown]
